### ISSUE TITLE
`git-hook/pre-push: exec: python: not found`

### ISSUE THREAD

**Issue by JordanLongstaff** [microsoft/vscode#193634]
Does this issue occur when all extensions are disabled?: N/A

- VS Code Version: 1.82.2
- OS Version: macOS Ventura 13.5.2

I have a pre-push hook that runs a Python program:
```
#!/usr/bin/env bash
# The devcheck has become a blackbox taking too much responsibility. Consider a pre-commit
# before adding more complexity to the devcheck.
exec python -m server.devcheck pre-push
```

The Python executable definitely exists on my computer, yet this stops me from being able to push commits using VS Code. Output:
```
2023-09-20 19:23:47.810 [info] > git push -u origin branch-name-redacted [957ms]
2023-09-20 19:23:47.810 [info] git-hooks/pre-push: line 4: exec: python: not found
error: failed to push some refs to 'github.com:EncircleInc/encircle.git'
```

This has to be an issue with VS Code, because I can run the above `git push -u ...` command from the CLI and it works with no problems. Whatever environment VS Code runs git in, it's not one that recognizes the executables in my shell.

Either this is a bug, or I need to configure my settings somehow. If it's the latter, please offer whatever help you can.

Steps to Reproduce:

1. Write any Python program you wish (it could be just a "Hello, world" program) and write a pre-push hook to run it like the above example.
2. Make a branch and add a commit to it.
3. Try publishing the branch through VS Code. The expectation is that it runs `git push -u` and successfully pushes the branch to the remote. The actual behaviour is that it fails to find Python.

**Comment by lszomoru** [comment]
@brettcannon, does this one ring a bell? Thanks!

**Comment by brettcannon** [comment]
Nope, this isn't a specific issue that I'm aware of if Python is globally installed and accessible on `PATH` (assuming that's how VS Code executes pre-commit hooks).

**Comment by bahung1221** [comment]
Same issue when using git in vscode on macOS, it worked when using git in terminal but not in vscode. (It previously worked but the issue happened recently)

<img width="1064" alt="Screenshot 2023-12-15 at 09 47 04" src="https://github.com/microsoft/vscode/assets/27690861/a4a68e00-7907-4205-87f0-b014f7581f99">

**Comment by Link512** [comment]
I'm currently experiencing the same(ish) issue with a golang repo. I have a pre-commit hook that runs a linter. If I do `git commit` from the terminal it works, however trying to commit from vscode gives me the following

```
> git -c user.useConfigOnly=true commit --quiet --allow-empty-message --file - -S
/home/<SNIP>/dotgit/hooks/pre-commit-format-lint:7: command not found: goimports
/home/<SNIP>/dotgit/hooks/pre-commit-format-lint:23: command not found: golangci-lint
```

Somehow, when doing a commit via vscode, my shell's `PATH` variable isn't read. I've had this issue in the past, but then got fixed when I updated vscode. Now it was reintroduced with the February `1.87` update.

**Comment by lszomoru** [comment]
How do you launch VS Code? Are you launching it from the Terminal by running `code` or from the dock? If you are running it from the dock you are hitting the issue that is documented [here](https://code.visualstudio.com/docs/supporting/FAQ#_resolving-shell-environment-fails). Please use the troubleshooting steps from the documentation to further investigate the issue.
